File: 1erdepartamental/c++/8.promedio.cpp
// ============================================
// PROGRAMA: Estadísticas con Parámetros Fijos
// DESCRIPCIÓN: Calcula suma, promedio, máximo, mínimo
// ============================================

#include <iostream>
#include <array>
using namespace std;


/**
 * Estadísticas sobre 5 números fijos
 * 
 * Implementaciones manuales sin usar algorithms
 */
class Estadisticas {
private:
    array<double, 5> __numeros;

public:
    Estadisticas(double n1, double n2, double n3, double n4, double n5) {
        __numeros = {n1, n2, n3, n4, n5};
    }
    
    // Getter que retorna copia
    array<double, 5> getNumeros() const {
        return __numeros;
    }
    
    double suma() const {
        double total = 0;
        for (double n : __numeros) {
            total += n;
        }
        return total;
    }
    
    double promedio() const {
        return suma() / __numeros.size();
    }
    
    double maximo() const {
        double m = __numeros[0];
        for (double n : __numeros) {
            if (n > m) m = n;
        }
        return m;
    }
    
    double minimo() const {
        double m = __numeros[0];
        for (double n : __numeros) {
            if (n < m) m = n;
        }
        return m;
    }
};


// ============================================
// EJECUCIÓN
// ============================================
int main() {
    Estadisticas e(10, 25, 5, 30, 15);
    
    cout << "Suma: " << e.suma() << endl;
    cout << "Promedio: " << e.promedio() << endl;
    cout << "Máximo: " << e.maximo() << endl;
    cout << "Mínimo: " << e.minimo() << endl;
    
    return 0;
}

File: 1erdepartamental/python/8.promedio.py
# ============================================
# PROGRAMA: Estadísticas con Parámetros Fijos
# DESCRIPCIÓN: Calcula suma, promedio, máximo, mínimo
# ============================================


class Estadisticas:
    """
    Estadísticas sobre 5 números fijos.

    Implementaciones manuales (sin usar sum(), max(), min())
    """

    def __init__(self, n1, n2, n3, n4, n5):
        self.__numeros = [n1, n2, n3, n4, n5]

    @property
    def numeros(self):
        return self.__numeros.copy()

    def suma(self):
        """Suma manual con bucle"""
        total = 0
        for n in self.__numeros:
            total += n
        return total

    def promedio(self):
        """Media aritmética"""
        return self.suma() / len(self.__numeros)

    def maximo(self):
        """Encuentra el mayor sin usar max()"""
        m = self.__numeros[0]
        for n in self.__numeros:
            if n > m:
                m = n
        return m

    def minimo(self):
        """Encuentra el menor sin usar min()"""
        m = self.__numeros[0]
        for n in self.__numeros:
            if n < m:
                m = n
        return m


# ============================================
# EJECUCIÓN
# ============================================
if __name__ == "__main__":
    e = Estadisticas(10, 25, 5, 30, 15)

    print(f"Números: {e.numeros}")
    print(f"Suma: {e.suma()}")
    print(f"Promedio: {e.promedio()}")
    print(f"Máximo: {e.maximo()}")
    print(f"Mínimo: {e.minimo()}")
